docs/source/reconciliation.rst

```diff
@@ -0,0 +1,134 @@
+Reconciliation
+==============
+Reconciliation is the high-level process of modifying an (extended) prefix map with
+domain-specific rules. This is important as it allows for building on existing
+(extended) prefix maps without having to start from scratch. Further, storing the
+rules to transform an existing prefix map allows for high-level discussion about
+the differences and their reasons.
+
+As a specific example, the `Bioregistry <https://bioregistry.io/>`_ uses ``snomedct`` as a preferred prefix for
+the Systematized Nomenclature of Medicine - Clinical Terms (SNOMED-CT). The
+OBO Foundry community prefers to use ``SCTID`` as the preferred prefix for this
+resource. Rather than maintaining a different extended prefix map than the Bioregistry,
+the OBO Foundry community could enumerate its preferred modifications to the base
+(extended) prefix map, then create its prefix map by transforming the Bioregistry's.
+
+Similarly, a consumer of the OBO Foundry prefix map who's implementing a resolver might want to override the URI prefix
+associated with the `Ontology of Vaccine Adverse Events (OVAE) <https://bioregistry.io/registry/ovae>`_
+to point towards the Ontology Lookup Service instead of the default OntoBee.
+
+There are two operations that are useful for transforming an existing (extended) prefix
+map:
+
+1. **Remapping** is when a given CURIE or URI prefix is replaced with another. See :func:`curies.remap`.
```

Comment:
### Alternative terminology
I kind of like these terms "remapping" and "rewiring", but just want to alter alternatives:
- Remapping --> "CURIE prefix remapping"
- Rewiring --> "URI prefix remapping"

Sounds more boring, but less terminology to learn.

### Accuracy of terminology
Also, the terms "remap" or "rewire" may technically be incorrect here, because there are some edge cases, such as  rewiring case "1. CURIE prefix doesn't exists", where nothing is getting re-mapped/re-wired. Simply a new entry is being added.

Reply:
that's a good point. we might not want to allow this operation to add new content, since this is more of the job of chaining converters or directly adding content

Reply:
After a second thought, I have removed the functionality where it adds to the prefix map. It's better for the vocabulary to be consistent and accurate

Reply:
I also added in the corresponding URI remapping functionality to make more clear that rewiring is not URI prefix remapping